Fix template configuration of snippets index results view

fidoriel opened the pull request:
Fixes an error where an exception is thrown, because the results only flag is True on Ajax search requests, but the 
results template is none.



_Please check the following:_

-   [x] Do the tests still pass?[^1]
-   [x] Does the code comply with the style guide?
    -   [x] Run `make lint` from the Wagtail root.
-   [ ] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [x] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [x] **Please list the exact browser and operating system versions you tested**
    -  I have tested firefox, with different screen sizes on manjaro. This PR does not modify any frontend code, so testing that is not needed.
    -   [ ] **Please list which assistive technologies [^3] you tested**:
-   [ ] For new features: Has the documentation been updated accordingly?

Reproduce:
Create an indexed Snippet and register it. Add some entries to the model and perform a search. The template for the results is not found, render is performed on None. This PR sets the correct template to resolve this issue.

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/core/handlers/base.py", line 220, in _get_response
    response = response.render()
               ^^^^^^^^^^^^^^^^^
  File "/code/wagtail/wagtail/admin/auth.py", line 204, in overridden_render
    return render()
           ^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/template/response.py", line 111, in render
    self.content = self.rendered_content
                   ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/django/template/response.py", line 89, in rendered_content
    return template.render(context, self._request)
           ^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'render'
```

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

lb- commented:
@fidoriel thank you, can you please add unit tests and also link this PR (in the description) with the issue being fixed. 

Or if no issue, please flesh out the description with steps to reproduce.

fidoriel commented:
@lb- I dont know why the tests fail here, on my fork do they work fine. Segfault somewhere in postgres. When looking back in to the logic for figuring out how to test it, I was wondering about it and changed the PR. It seems if the logic I removed was no longer used, default false, only one place was setting it to true, causing the None-Template used.  The super templates are correct, the remplate needed.

gasman commented:
Thanks @fidoriel! I can confirm I'm seeing the same issue when testing against [bakerydemo](https://github.com/wagtail/bakerydemo) (running a search from the index page of the Person snippet), and the fix looks good to me.

@laymonage Please can you double-check this fix? It all checks out to me (the 'index results view' is just the standard IndexView with the template swapped out, and the viewset handles that by passing in an alternative `template_name`, so the logic with `results_template_name` and the `results_only` flag is redundant), but I just want to be sure I've not missed anything relating to the customisability of these templates.

laymonage commented:
@gasman yep I'm aware of this and it's on my list for today. I'll try to add a test to ensure this is all good and merge it afterwards

laymonage commented:
Thanks @fidoriel, the fix looks good to me and I've added tests to ensure the bug has been fixed. Nice to see that the fix is actually to *remove* code!

I've credited you as `fidoriel` as I couldn't find any other name. If you would like me to change it, please let me know. Again, thanks!
